orange/src/test/java/com/hartwig/hmftools/orange/algo/purple/ChromosomalRearrangementsDeterminerTest.java

```diff
@@ -0,0 +1,103 @@
+        assertFalse(chromosomalRearrangementsDeterminer.determine1qTrisomy(without1qTrisomy));
+    }
+
+    @Test
+    public void canDetermine1p19qCoDeletion()
+    {
+        int centromere1 = refCoordinates.centromere("1");
+        int centromere19 = refCoordinates.centromere("19");
+        int endChromosome19 = refCoordinates.length("19");
+
+        int stepSize1 = centromere1 / 4;
+        List<Integer> steps1 = createSteps(stepSize1, 1, centromere1 + 3000000);
+
+        int stepSize19 = (endChromosome19 - centromere19) / 4;
+        List<Integer> steps19 = createSteps(stepSize19, centromere19 - 3000000, endChromosome19);
+
+        List<PurpleCopyNumber> arm1p = Lists.newArrayList(copyNumber("1", 1, steps1.get(1), 0.1), copyNumber("1",
+                steps1.get(1) + 1, steps1.get(2), 0.15), copyNumber("1", steps1.get(2) + 1, steps1.get(3), 0.1));
+        List<PurpleCopyNumber> arm19q = Lists.newArrayList(copyNumber("19", steps19.get(0), steps19.get(1), 0.1), copyNumber("19",
+                steps19.get(1) + 1, steps19.get(2), 0.15), copyNumber("19", steps19.get(2) + 1, steps19.get(3), 0.1));
+
+        List<PurpleCopyNumber> withDeletion1p = new ArrayList<>(arm1p);
+        withDeletion1p.add(copyNumber("1", steps1.get(3) + 1, centromere1 + 3000000, 0.1));
+        List<PurpleCopyNumber> withoutDeletion1p = new ArrayList<>(arm1p);
+        withoutDeletion1p.add(copyNumber("1", steps1.get(3) + 1, centromere1 + 3000000, 0.9));
+
+        List<PurpleCopyNumber> withDeletion19q = new ArrayList<>(arm19q);
+        withDeletion19q.add(copyNumber("19", steps19.get(3) + 1, endChromosome19, 0.1));
+        List<PurpleCopyNumber> withoutDeletion19q = new ArrayList<>(arm19q);
+        withoutDeletion19q.add(copyNumber("19", steps19.get(3) + 1, endChromosome19, 0.9));
+
+        final List<PurpleCopyNumber> with1pDeletionAndWith19QDeletion = Lists.newArrayList(withDeletion1p);
+        with1pDeletionAndWith19QDeletion.addAll(withDeletion19q);
+        final List<PurpleCopyNumber> with1pDeletionAndWithout19QDeletion = Lists.newArrayList(withDeletion1p);
+        with1pDeletionAndWithout19QDeletion.addAll(withoutDeletion19q);
+        final List<PurpleCopyNumber> without1pDeletionAndWith19QDeletion = Lists.newArrayList(withoutDeletion1p);
+        without1pDeletionAndWith19QDeletion.addAll(withDeletion19q);
+
+        assertTrue(chromosomalRearrangementsDeterminer.determine1p19qCodeletion(with1pDeletionAndWith19QDeletion));
+        assertFalse(chromosomalRearrangementsDeterminer.determine1p19qCodeletion(with1pDeletionAndWithout19QDeletion));
+        assertFalse(chromosomalRearrangementsDeterminer.determine1p19qCodeletion(without1pDeletionAndWith19QDeletion));
+    }
+
+    @NotNull
+    private static PurpleCopyNumber copyNumber(String chromosome, int start, int end, double copyNumber)
+    {
+        return PurpleTestUtils.createCopyNumber(chromosome, start, end, copyNumber).build();
```

Comment:
Don't think I'd have this function since the PurpleTestUtils function is already quite concise, but fairly minor and not super strong opinion

Reply:
I do like it :)